src/server/pfs/pfs.go

```diff
@@ -429,7 +429,7 @@ func IsDropWithChildrenErr(err error) bool {
 func ValidateSQLDatabaseEgress(sql *pfs.SQLDatabaseEgress) error {
 	if sql != nil {
```

Comment:
Outside the scope this PR, but I'd prefer returning to indenting, i.e.

```go
if sql == nil {
    return nil
}
secret := sql.GetSecret()
if secret == nil {
   return errors.Errorf("egress.sql_database.secret is required")
}
// ...
```

This is not quite the rule: https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow but "Try to keep the normal code path at a minimal indentation" is good advice.

Reply:
Agreed, this way is cleaner. I normally structure my code like this when dealing with errors, but in this case it's kind of the opposite where a nil value results in error.